src/main/java/se/liu/ida/hefquin/engine/data/VocabularyMapping.java

```diff
@@ -1,8 +1,25 @@
 package se.liu.ida.hefquin.engine.data;
 
+import java.util.Set;
+import org.apache.jena.reasoner.TriplePattern;
+
 
-//TODO: Should this be extends Model?
 public interface VocabularyMapping {
 
-	// TODO: What should this contain?
+	/**
+	 * Applies this vocabulary mapping to the given triple pattern and
+	 * returns the resulting triple patterns. If this mapping is not relevant
+	 * for the given triple pattern (i.e., applying the mapping to the given
+	 * triple pattern does not have any effect), then the result of this
+	 * function is a singleton set that simply contains the given triple
+	 * pattern.
+	 */
+	
+	/**
+	 * TODO: Is TripplePattern enough? Which data structure to use?
+	 * Unions and intersections also need to be represented
+	 * Would Graph be a fitting? Both Jena union and intersection are graphs
+	 */
+		
+	Set<TriplePattern> translateTriple( TriplePattern tp );
```

Comment:
Should I remove the implementation of the translation function for now, and create a new branch and PR for it once the VocabularyMapping looks good?

Reply:
Good point Sebastian!

Juliette, yes, remove the implementation of the translation function from this PR. More precisely, save it somewhere else so that you can use it for the next PR, and in this PR simply `return null` in this function.